src/components/pdf-upload/PdfUploadAgent.tsx

```diff
@@ -39,34 +39,51 @@ export const PdfUploadAgent = ({
 
       const apiBaseUrl =
         import.meta.env.VITE_API_URL || "http://localhost:8787";
-      const response = await fetch(`${apiBaseUrl}/pdf/authenticate`, {
-        method: "POST",
-        headers: {
-          "Content-Type": "application/json",
-        },
-        body: JSON.stringify({
-          sessionId,
-          providedKey: "check-status-only",
-        }),
-      });
 
-      const result = (await response.json()) as {
-        success: boolean;
-        authenticated: boolean;
-        error?: string;
-      };
+      // Add timeout to prevent hanging
+      const controller = new AbortController();
+      const timeoutId = setTimeout(() => controller.abort(), 10000); // 10 second timeout
 
-      if (response.ok && result.success && result.authenticated) {
-        setIsAuthenticated(true);
-        setAuthError(null);
-        setShowAuthInput(false);
+      // Use the is-session-authenticated endpoint for status checking
+      const response = await fetch(
+        `${apiBaseUrl}/pdf/is-session-authenticated?sessionId=${sessionId}`,
+        {
+          method: "GET",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          signal: controller.signal,
+        }
+      );
+
+      clearTimeout(timeoutId);
```

Comment:
i don't super understand how the `clearTimeout` works here - what would be the error response if things go south? do we retry on timeout?